packages/utils/src/carequality/cq-monthly-stats/shared.ts

```diff
@@ -0,0 +1,325 @@
+  lat?: number;
+  lon?: number;
+  state?: string;
+  data?: Organization;
+  point?: string;
+  managingOrganization?: string;
+  managingOrganizationId?: string;
+  gateway: boolean;
+  active: boolean;
+  lastUpdatedAtCQ: string;
+};
+
+export type RequestParams = {
+  cqDirectory: CQDirectoryEntryData[];
+  endOfPreviousMonth: string;
+  dayIndex: number;
+};
+
+export type GWStats = {
+  nonErroredResponses?: number;
+  totalDocRetrieved?: number;
+  avgResponseTimeMs?: number;
+};
+
+export type GWWithStats = {
+  gwId: string;
+} & GWStats;
+
+export type Implementer = {
+  implementerId: string;
+  implementerName: string;
+};
+
+export type ImplementerWithAvgResp = Implementer & {
+  avgResponseTimeMs: number;
+};
+
+export type ImplementerWithGwStats = Implementer & {
+  gwStats: GWWithStats[];
+};
+
+export type ImplementerStats = Implementer & GWStats;
+
+export type ImplementerStatsByDay = {
+  [day: string]: ImplementerWithGwStats[];
+};
+
+export type DurationKey =
+  | "xcpdAvgResponseTimeMs"
+  | "xcaDQAvgResponseTimeMs"
+  | "xcaDRAvgResponseTimeMs";
+
+export type DurationAvgsArray = {
+  [key in DurationKey]?: number[];
+};
+
+export type DurationAvgs = {
+  [key in DurationKey]?: number;
+};
+
+export type MonthlyImplementerStats = {
+  year: number;
+  month: number;
+} & ImplementerStats &
+  DurationAvgsArray;
+
+export type CountPerGW = { [key: string]: [number] };
+
+export type DailyAvgsByImplementer = {
+  [implementerId: string]: {
+    implementerId: string;
+    implementerName: string;
+  } & DurationAvgsArray;
+};
+
+export type MonthlyAvgByImplementer = {
+  [implementerId: string]: {
+    implementerId: string;
+    implementerName: string;
+  } & DurationAvgs;
+};
+
+export type QueryResultsTableNames = "document_query_result" | "document_retrieval_result";
+
+export const QUERY_RESULTS_TABLE_NAMES = {
+  documentQuery: "document_query_result",
+  documentRetrieval: "document_retrieval_result",
+} as const;
+
+export async function queryResultsTable<TableResults>(
+  tableName: QueryResultsTableNames,
+  endOfPreviousMonth: string,
+  dayIndex: number
+): Promise<TableResults[]> {
+  const dayPlusOne = dayIndex + 1;
+
+  console.log(`Querying table ${tableName} for day ${dayPlusOne}...`);
+
+  const query = `
+      SELECT * FROM ${tableName}
+      WHERE created_at > date '${endOfPreviousMonth}' - INTERVAL '${dayPlusOne} days'
+      AND created_at < date '${endOfPreviousMonth}' - INTERVAL '${dayPlusOne - 1} days'
+    `;
+
+  const results = await readOnlyDBPool.query(query, {
+    type: QueryTypes.SELECT,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const resultsData = results.map((result: any) => result.data) as TableResults[];
+
+  console.log(`Results for table ${tableName} day ${dayPlusOne}:`, results.length);
+
+  return resultsData;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getDurationsPerGW(results: any[]): GWWithStats[] {
```

Comment:
Can't we type `results` to `TableResults[]`?

Reply:
The results from athena are different than our db